src/mixins/UserHasPermission.js

```diff
@@ -0,0 +1,26 @@
+import { mapState } from "vuex";
+
+const UserHasPermission = {
+	props: {
+		permission: {
+			type: [String, Function],
+			default: () => () => false,
+		},
+	},
+	computed: {
+		...mapState({
+			userPermissions: (state) =>
+				state && state.auth && state.auth.user && state.auth.user.permissions
+					? state.auth.user.permissions.map((p) => p.toLowerCase())
+					: [],
+		}),
+		$_hasPermission() {
+			return typeof this.permission === "string"
+				? !this.permission ||
+						this.userPermissions.includes(this.permission.toLowerCase())
+				: !this.permission() || this.permission(this.userPermissions);
+		},
+	},
+};
+
+export default UserHasPermission;
```

Comment:
```suggestion
export default {
	props: {
		permission: {
			type: [String, Function],
			default: () => () => false,
		},
	},
	computed: {
		...mapState({
			userPermissions: (state) =>
				state && state.auth && state.auth.user && state.auth.user.permissions
					? state.auth.user.permissions.map((p) => p.toLowerCase())
					: [],
		}),
		$_hasPermission() {
			return typeof this.permission === "string"
				? !this.permission ||
						this.userPermissions.includes(this.permission.toLowerCase())
				: !this.permission() || this.permission(this.userPermissions);
		},
	},
};
```

to keep the codebase consistent you could directly export the mixin instead of storing it in a variable before.